bundles/org.openhab.binding.wundergroundupdatereceiver/src/main/java/org/openhab/binding/wundergroundupdatereceiver/internal/WundergroundUpdateReceiverHandler.java

```diff
@@ -0,0 +1,176 @@
+            @Nullable
+            Map<String, String[]> requestParameters = discoveryService.getUnhandledStationRequest(stationId);
+            if (requestParameters != null) {
+                requestParameters.forEach((String parameter, String[] query) -> {
+                    WundergroundUpdateReceiverParameterMapping channelTypeMapping = WundergroundUpdateReceiverParameterMapping
+                            .getOrCreateMapping(parameter, String.join("", query), channelTypeProvider);
+                    ChannelBuilder channelBuilder = ChannelBuilder.create(
+                            new ChannelUID(thing.getUID(), channelTypeMapping.channelGroup, parameter),
+                            channelTypeRegistry.getChannelType(channelTypeMapping.channelTypeId).getItemType());
+                    thingBuilder.withChannel(channelBuilder.build());
+                });
+            }
+            updateThing(thingBuilder.build());
+        }
+        this.wundergroundUpdateReceiverServlet = Objects.requireNonNull(wunderGroundUpdateReceiverServlet);
+        if (stationId != null) {
+            discoveryService.removeUnhandledStationId(stationId);
+        }
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        logger.trace("Ignoring command {}", command);
+    }
+
+    @Override
+    public void handleConfigurationUpdate(Map<String, Object> configurationParameters) {
+        super.handleConfigurationUpdate(configurationParameters);
+        this.wundergroundUpdateReceiverServlet.handlerConfigUpdated(this);
+    }
+
+    @Override
+    public void initialize() {
+        this.config = Objects.requireNonNull(getConfigAs(WundergroundUpdateReceiverConfiguration.class));
+        wundergroundUpdateReceiverServlet.addHandler(this);
+        if (wundergroundUpdateReceiverServlet.isActive()) {
+            updateStatus(ThingStatus.ONLINE);
+            logger.debug("Wunderground update receiver listening for updates to station id {}", config.stationId);
+            return;
+        }
+        updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.HANDLER_INITIALIZING_ERROR,
+                wundergroundUpdateReceiverServlet.getErrorDetail());
+    }
+
+    @Override
+    public void dispose() {
+        wundergroundUpdateReceiverServlet.removeHandler(this.getStationId());
+        super.dispose();
+        logger.debug("Wunderground update receiver stopped listening for updates to station id {}", config.stationId);
+    }
+
+    public void updateChannelStates(Map<String, String> requestParameters) {
+        requestParameters.forEach(this::updateChannelState);
+        updateState(lastReceivedChannel, new DateTimeType());
+        String lastQuery = requestParameters.getOrDefault(LAST_QUERY, "");
+        if (lastQuery.isEmpty()) {
+            return;
+        }
+        updateState(queryStateChannel, StringType.valueOf(lastQuery));
+        triggerChannel(queryTriggerChannel, lastQuery);
```

Comment:
Is there any reason to feed the last query also to a stateful channel?

Reply:
You need it to re-send the measurements to the actual wunderground.com update url.

Reply:
There is an example in the README. Re-creating the url from the other channels is not easy as the formatting has to be exactly as in the originale url.